torchmetrics/audio/pit.py

```diff
@@ -0,0 +1,115 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable, Dict, Optional
+
+from torch import Tensor, tensor
+
+from torchmetrics.functional.audio.pit import pit
+from torchmetrics.metric import Metric
+
+
+class PIT(Metric):
+    """ Permutation invariant training (PIT). The PIT implements the famous Permutation Invariant Training method in
+    speech separation field in order to calculate audio metrics in a permutation invariant way.
+
+    Forward accepts
+
+    - ``preds``: ``shape [batch, spk, ...]``
+    - ``target``: ``shape [batch, spk, ...]``
+
+    Args:
+        metric_func:
+            a metric function accept a batch of target and estimate, i.e. metric_func(target[:, i, ...],
+            estimate[:, j, ...]), and returns a batch of metric tensors [batch]
+        eval_func:
+            the function to find the best permutation, can be 'min' or 'max', i.e. the smaller the better
+            or the larger the better.
+        compute_on_step:
+            Forward only calls ``update()`` and returns None if this is set to False. default: True
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When `None`, DDP
+            will be used to perform the allgather.
+        kwargs:
+            additional args for metric_func
+
+    Returns:
+        average PIT metric
+
+    Example:
+        >>> import torch
+        >>> from torchmetrics import PIT
+        >>> from torchmetrics.functional.audio import si_snr
+        >>> preds = torch.randn(3, 2, 5) # [batch, spk, time]
+        >>> target = torch.randn(3, 2, 5) # [batch, spk, time]
+        >>> pit = PIT(si_snr, 'max')
+        >>> avg_pit_metric = pit(preds, target)
```

Comment:
```suggestion
        >>> pit(preds, target)
```
of what is wrong with the number?